src/shipit_code_coverage/shipit_code_coverage/github.py

```diff
@@ -43,30 +43,24 @@ def post_github_status(self, commit_sha):
     def get_commit(self, mercurial_commit):
         def get_commit():
             r = requests.get('https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s' % mercurial_commit)
+            r.raise_for_status()
```

Comment:
If you throw the right exception directly here (`raise Exception('Mercurial commit is not available yet on mozilla/gecko-dev.')`), you can avoid wrapping the retry call in a try except

Reply:
Done in f259df0